### ISSUE TITLE
keras.ops.sort(axis=None) fails on eager tensors instead of flattening and sorting

### ISSUE THREAD

**Issue by AGFACBNNR** [keras-team/keras#22543]
### Summary
`keras.ops.sort` with `axis=None` is documented to flatten the tensor before sorting, but on TensorFlow eager tensors it raises an error.

### Reproduction
```python
import keras
import tensorflow as tf

x = tf.constant([[[1, 4, 3], [9, 7, 6]]], dtype=tf.float32)
print(keras.ops.sort(x, axis=None))
```

### Expected result
The tensor should be flattened and returned in ascending order.

### Actual result
An exception is raised:

`Attempt to convert a value (None) with an unsupported type (<class 'NoneType'>) to a Tensor.`

### Notes
The same API call with a `keras.Input` placeholder does not raise this error, so the issue appears specific to eager execution.

**Comment by whitakercaleb09-dev** [comment]
Thank you for reporting this issue with keras.ops.sort.

**Comment by dhantule** [comment]
Hi @AGFACBNNR, thanks for reporting this. 

I have tested this with `keras-nightly`, it seems to be fixed. Please refer this attached [gist](https://colab.sandbox.google.com/gist/dhantule/63ecc689fb769ffed60eac3a9f92562f/keras-ops-sort-axis-none-fails-on-eager-tensors-instead-of-flattening-and-sorting-22543.ipynb).

**Comment by divyashreepathihalli** [comment]
@keras-team /triage

**Comment by divyashreepathihalli** [comment]
@keras-team /triage

**Comment by Saumay** [comment]
Looks like this is already fixed on current master (e94cb07d5) by #22544. Running `keras.ops.sort(tf.constant([[1.,4.,3.]]), axis=None)` now returns a flattened sorted tensor of shape `(3,)` as expected. Probably safe to close.

**Comment by pctablet505** [comment]
Fixed on master. The TensorFlow backend sort implementation reshapes to [-1] and sets axis=0 when axis=None. Verified in backend/tensorflow/numpy.py and ops/numpy_test.py.
